Modules/_struct.c

```diff
@@ -1931,14 +1931,40 @@ s_pack_into(PyObject *self, PyObject **args, Py_ssize_t nargs, PyObject *kwnames
     }
 
     /* Support negative offsets. */
-    if (offset < 0)
+    if (offset < 0) {
+         /* Check that negative offset is low enough to fit data */
+        if (offset + soself->s_size > 0) {
+            PyErr_Format(StructError,
+                         "No space to pack %zd bytes at offset %zd",
```

Comment:
I admit that we are pretty inconsistent with the styles of our exception messages, but I think these two should start with lowercase letters.